packages/test/test-end-to-end-tests/src/test/entryPointCompat.spec.ts

```diff
@@ -0,0 +1,118 @@
+
+describe("entryPoint compat", () => {
+	let provider: ITestObjectProvider;
+
+	class TestDataObject extends DataObject {
+		public get _root() {
+			return this.root;
+		}
+		public get _context() {
+			return this.context;
+		}
+	}
+
+	async function getDefaultFluidObject(runtime: IContainerRuntime): Promise<FluidObject> {
+		// eslint-disable-next-line @typescript-eslint/no-non-null-assertion
+		return (await runtime.getAliasedDataStoreEntryPoint?.("default"))!.get();
+	}
+
+	async function createRequestOnlyContainer(): Promise<IContainer> {
+		const dataObjectFactory = new DataObjectFactory("TestDataObject", TestDataObject, [], []);
+		const runtimeFactory = new ContainerRuntimeFactoryWithDefaultDataStore(dataObjectFactory, [
+			[dataObjectFactory.type, Promise.resolve(dataObjectFactory)],
+		]);
+
+		return provider.createContainer(runtimeFactory);
+	}
+
+	async function createContainer(): Promise<IContainer> {
+		const dataObjectFactory = new DataObjectFactory("TestDataObject", TestDataObject, [], []);
+		const runtimeFactory = new ContainerRuntimeFactoryWithDefaultDataStore(
+			dataObjectFactory,
+			[[dataObjectFactory.type, Promise.resolve(dataObjectFactory)]],
+			undefined,
+			undefined,
+			undefined,
+			async (runtime: IContainerRuntime) => getDefaultFluidObject(runtime),
+		);
+
+		return provider.createContainer(runtimeFactory);
+	}
+
+	describeNoCompat("no compat", (getTestObjectProvider) => {
+		beforeEach(async () => {
+			provider = getTestObjectProvider();
+		});
+
+		it("entryPoint pattern", async () => {
+			const container = await createContainer();
+			const entryPoint = await container.getEntryPoint?.();
+			assert.notStrictEqual(entryPoint, undefined, "entryPoint was undefined");
+		});
+
+		it("request pattern", async () => {
+			const container = await createRequestOnlyContainer();
+			const requestResult = await container.request({ url: "/" });
+
+			assert.strictEqual(requestResult.status, 200, requestResult.value);
+			assert.notStrictEqual(requestResult.value, undefined, "requestResult was undefined");
+		});
+
+		it("both entryPoint and request pattern", async () => {
+			const container = await createContainer();
+			const entryPoint = await container.getEntryPoint?.();
+			const requestResult = await container.request({ url: "/" });
+
+			assert.strictEqual(requestResult.status, 200, requestResult.value);
+			assert.strictEqual(
+				entryPoint,
+				requestResult.value,
+				"entryPoint and requestResult expected to be the same",
+			);
+		});
+	});
+
+	// Simulating old loader code
+	describeLoaderCompat("loader compat", (getTestObjectProvider) => {
+		beforeEach(async () => {
+			provider = getTestObjectProvider();
+		});
+
+		it("something", () => {
+			// have unit test in containerRuntime.spec.ts to verify you can use both load and loadRuntime
+		});
+
+		it("request pattern works", async () => {
+			const container = await createRequestOnlyContainer();
```

Comment:
Trying to understand what this is testing... with "Loader Compat", the Provider will use old Loader code but new Runtime code. But there's no recent changes to FF Loader bits that matter, are there?

In other words, when we speak of a 1P partner's "old loader" bits, we mean their old host code that is going to call Container.request.  Not so much that the Loader code itself is old.

So what's the value prop of these LoaderCompat tests?